### PR TITLE
Features/7575 - Initial work on Protected word response

### PR DESCRIPTION
**Pull request by sslaws**
Plumbing work that is ready for merge and non-breaking.

### CODE REVIEW

**Comment by healthgatewaybc** [comment]
Immunization Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=immunization_pr_181)

**Comment by healthgatewaybc** [comment]
Common Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=commonlibrary_pr_181)

**Comment by healthgatewaybc** [comment]
Patient Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=patientservice_pr_181)

**Comment by healthgatewaybc** [comment]
Common Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=commonlibrary_pr_181)

**Comment by healthgatewaybc** [comment]
Patient Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=patientservice_pr_181)

**Comment by healthgatewaybc** [comment]
Immunization Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=immunization_pr_181)

**Comment by TiagoGraf89** [comment]
Apparently there are some tests broken

File: Apps/Medication/src/Controllers/MedicationStatementController.cs [hunk 1]
```diff
@@ -75,38 +75,26 @@ public MedicationStatementController(IAuthorizationService authorizationService,
...
         {
             ClaimsPrincipal user = this.httpContextAccessor.HttpContext.User;
             var isAuthorized = await this.authorizationService.AuthorizeAsync(user, hdid, PolicyNameConstants.UserIsPatient).ConfigureAwait(true);
             if (!isAuthorized.Succeeded)
             {
-                return new ChallengeResult();
+               return new ForbidResult();
             }
-
+            RequestResult<List<MedicationStatement>> result = new RequestResult<List<MedicationStatement>>();
```
Comment (FuriousLlama) [inline Apps/Medication/src/Controllers/MedicationStatementController.cs]: nit lets inline the initialization of ResultStatus and ResultMessage

File: Apps/Medication/src/Controllers/PharmacyController.cs [hunk 2]
```diff
@@ -59,28 +59,17 @@ public PharmacyController(IPharmacyService pharmacyService)
         public async Task<RequestResult<Pharmacy>> GetPharmacy(string pharmacyId)
         {
             HNMessage<Pharmacy> pharmacyResult = await this.pharmacyService.GetPharmacyAsync(pharmacyId).ConfigureAwait(true);
-
-            if (pharmacyResult.IsError)
+            RequestResult<Pharmacy> result = new RequestResult<Pharmacy>();
```
Comment (FuriousLlama) [inline Apps/Medication/src/Controllers/PharmacyController.cs]: nit: lets pass the parameters for initialization
```
RequestResult<Pharmacy> result = new RequestResult<Pharmacy>()
{
 ResultStatus = pharmacyResult.Result,
 ResultMessage = pharmacyResult.ResultMessage
}

File: Apps/Medication/test/unit/Controllers/MedicationStatementController_Test.cs [hunk 3]
```diff
@@ -69,10 +69,18 @@ public async Task ShouldGetMedicationStatemets()
             MedicationStatementController controller = new MedicationStatementController(authzMock.Object, svcMock.Object, httpContextAccessorMock.Object);
 
             // Act
-            RequestResult<List<MedicationStatement>> actual = (RequestResult<List<MedicationStatement>>)await controller.GetMedicationStatements(hdid);
+            IActionResult actual = await controller.GetMedicationStatements(hdid);
+
+            Assert.IsType(typeof(JsonResult), actual);
```
Comment (FuriousLlama) [inline Apps/Medication/test/unit/Controllers/MedicationStatementController_Test.cs]: lets move the assertions to under the // Verify comment

File: Apps/WebClient/src/ClientApp/app/views/timeline.vue [hunk 4]
```diff
@@ -226,7 +226,7 @@ export default class TimelineComponent extends Vue {
       .getPatientMedicationStatements(this.user.hdid)
       .then(results => {
         console.log(results);
-        if (!results.errorMessage) {
+          if (results.resultStatus == 0) {
```
Comment (FuriousLlama) [inline Apps/WebClient/src/ClientApp/app/views/timeline.vue]: Instead of 0 we can use an Enum to type it

**Comment by healthgatewaybc** [comment]
Common Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=commonlibrary_pr_181)

**Comment by healthgatewaybc** [comment]
Patient Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=patientservice_pr_181)

**Comment by healthgatewaybc** [comment]
WebClient Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=webclient_pr_181)

**Comment by healthgatewaybc** [comment]
Immunization Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=immunization_pr_181)

File: Apps/Medication/src/Controllers/MedicationStatementController.cs (hunk 1 above)
Comment (sslaws) [reply Apps/Medication/src/Controllers/MedicationStatementController.cs]: Fixed

File: Apps/Medication/src/Controllers/PharmacyController.cs (hunk 2 above)
Comment (sslaws) [reply Apps/Medication/src/Controllers/PharmacyController.cs]: Fixed

File: Apps/Medication/test/unit/Controllers/MedicationStatementController_Test.cs (hunk 3 above)
Comment (sslaws) [reply Apps/Medication/test/unit/Controllers/MedicationStatementController_Test.cs]: I moved the location of the verify to include all of the assertions.

File: Apps/WebClient/src/ClientApp/app/views/timeline.vue (hunk 4 above)
Comment (sslaws) [reply Apps/WebClient/src/ClientApp/app/views/timeline.vue]: I've made this change.

**Comment by healthgatewaybc** [comment]
Immunization Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=immunization_pr_181)

**Comment by healthgatewaybc** [comment]
Common Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=commonlibrary_pr_181)

**Comment by healthgatewaybc** [comment]
Patient Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=patientservice_pr_181)

**Comment by healthgatewaybc** [comment]
Common Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=commonlibrary_pr_181)

**Comment by healthgatewaybc** [comment]
Immunization Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=immunization_pr_181)

**Comment by healthgatewaybc** [comment]
Patient Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=patientservice_pr_181)

**Comment by healthgatewaybc** [comment]
Common Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=commonlibrary_pr_181)

**Comment by healthgatewaybc** [comment]
Patient Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=patientservice_pr_181)

**Comment by healthgatewaybc** [comment]
Immunization Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=immunization_pr_181)

**Comment by healthgatewaybc** [comment]
Patient Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=patientservice_pr_181)

**Comment by healthgatewaybc** [comment]
Common Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=commonlibrary_pr_181)

**Comment by healthgatewaybc** [comment]
Immunization Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=immunization_pr_181)

**Comment by healthgatewaybc** [comment]
WebClient Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=webclient_pr_181)

**Comment by healthgatewaybc** [comment]
Patient Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=patientservice_pr_181)

**Comment by healthgatewaybc** [comment]
Common Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=commonlibrary_pr_181)

**Comment by healthgatewaybc** [comment]
Immunization Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=immunization_pr_181)

**Comment by healthgatewaybc** [comment]
Medication Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=Medicationservice_pr_181)

**Comment by healthgatewaybc** [comment]
WebClient Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=webclient_pr_181)

**Comment by healthgatewaybc** [comment]
Common Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=commonlibrary_pr_181)

**Comment by healthgatewaybc** [comment]
Patient Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=patientservice_pr_181)

**Comment by healthgatewaybc** [comment]
Immunization Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=immunization_pr_181)

**Comment by healthgatewaybc** [comment]
Medication Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=Medicationservice_pr_181)

**Comment by healthgatewaybc** [comment]
WebClient Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=webclient_pr_181)

**Pull request merged**